File: chat-service/src/main/java/com/sudo248/chatservice/repository/entity/ConversationInfo.java
package com.sudo248.chatservice.repository.entity;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Date;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class ConversationInfo {
    private String topic;
    private String conversationName;
    private String conversationImage;
    private String latestMessage;
    private long updateTime;
}

File: client/SocStaff/app/src/main/java/com/sudo248/soc_staff/domain/entity/chat/ConversationInfo.kt
package com.sudo248.soc_staff.domain.entity.chat

import com.sudo248.base_android.base.ItemDiff
import java.util.*

data class ConversationInfo(
    val topic: String,
    val conversationName: String,
    val conversationImage: String,
    val latestMessage: String,
    val updateTime: Long
) : ItemDiff {
    override fun isContentTheSame(other: ItemDiff): Boolean {
        return other is ConversationInfo && other == this
    }

    override fun isItemTheSame(other: ItemDiff): Boolean {
        return other is ConversationInfo && other.topic == topic && other.latestMessage == latestMessage && other.updateTime == updateTime
    }

}